src/btouch.cs

```diff
@@ -682,3 +625,40 @@ static void FixRelativeReferences (string sdk_path, List<string> references)
 		}
 	}
 }
+
+class SearchPathsAssemblyResolver : MetadataAssemblyResolver
+{
+	private readonly string[] libraryPaths;
+	private readonly string[] references;
+
+	public SearchPathsAssemblyResolver (string[] libraryPaths, string[] references)
+	{
+		this.libraryPaths = libraryPaths;
+		this.references = references;
+	}
+
+	public override Assembly? Resolve (MetadataLoadContext context, AssemblyName assemblyName)
+	{
+		string? name = assemblyName.Name;
+		if (name != null) {
+			foreach (var asm in context.GetAssemblies ()) {
+				if (asm.GetName ().Name == name)
+					return asm;
+			}
+
+			string dllName = name + ".dll";
+			foreach (var libraryPath in libraryPaths) {
+				string path = Path.Combine(libraryPath, dllName);
```

Comment:
Code style - space before opening parenthesis:

```suggestion
				string path = Path.Combine (libraryPath, dllName);
```